pkg/tuf/keys.go

```diff
@@ -89,77 +91,91 @@ func importRootKey(rootKey string, nRepo client.Repository, retriever notary.Pas
 		// Chooses the first root key available, which is initialization specific
 		// but should return the HW one first.
 		rootKeyID := rootKeyList[0]
-		log.Debugf("Signy found root key, using: %s\n", rootKeyID)
-
+		log.Debugf("found root key: %s\n", rootKeyID)
 		return []string{rootKeyID}, nil
 	}
 
 	return []string{}, nil
 }
 
+// Try to reuse a single targets/releases key across repositories.
+func reuseReleasesKey(r client.Repository) (data.PublicKey, error) {
+	// Get all known targets keys.
+	cryptoService := r.GetCryptoService()
+	keyList := cryptoService.ListKeys(releasesRoleName)
+
+	// Try to extract a single targets/releases key we can reuse.
+	switch len(keyList) {
+	case 0:
+		log.Debugf("No %s key available, need to make one", releasesRoleName)
+		return cryptoService.Create(releasesRoleName, r.GetGUN(), data.ECDSAKey)
```

Comment:
Is this creating a new key if one wasn't found? Is it persisted somewhere? Some more comments about what is happening here, and if it should ever be okay to happen outside of tests would be really helpful.